numpy/ma/core.py

```diff
@@ -2837,6 +2837,12 @@ def __new__(cls, data=None, mask=nomask, dtype=None, copy=False,
             _data = ndarray.view(_data, type(data))
         else:
             _data = ndarray.view(_data, cls)
+
+        # Handle the case where data is not a subclass of ndarray, but
+        # still has the _mask attribute like MaskedArrays
+        if hasattr(data, '_mask') and not isinstance(data, ndarray):
+            _data._mask = data._mask
+
```

Comment:
```suggestion
            # FIXME: should we set `_data._sharedmask = True`? 
```
Lets re-add that comment, just in case.

Reply:
:+1:, let me know if you want me to squash the commit out for cleanliness.